apps/antalmanac/src/stores/calendarizeHelpers.ts

```diff
@@ -200,3 +153,60 @@ export function parseDaysString(daysString: string): number[] {
 
     return days;
 }
+
+interface NormalizedWebSOCTime {
+    startTime: string;
+    endTime: string;
+}
+
+/**
+ * @param section
+ * @returns The start and end time of a course in a 24 hour time with a leading zero (##:##).
+ * @returns undefined if there is no WebSOC time (e.g. 'TBA', undefined)
+ */
+interface NormalizeTimeOptions {
+    timeIsTBA?: boolean;
+    startTime?: HourMinute | null;
+    endTime?: HourMinute | null;
+}
+
+/**
+ * @param section
+ * @returns The start and end time of a course in a 24 hour time with a leading zero (##:##).
+ * @returns undefined if there is no WebSOC time (e.g. 'TBA', undefined)
+ */
+export function normalizeTime(options: NormalizeTimeOptions): NormalizedWebSOCTime | undefined {
+    if (options.timeIsTBA || !options.startTime || !options.endTime) {
+        return;
+    }
+
+    // Times are normalized to ##:## (10:00, 09:00 etc)
+    const startHour = `${options.startTime.hour}`.padStart(2, '0');
+    const endHour = `${options.endTime.hour}`.padStart(2, '0');
+
+    const startTime = `${startHour}:${options.startTime.minute}`;
+    const endTime = `${endHour}:${options.endTime.minute}`;
+
+    return { startTime, endTime };
+}
+
+export function translate24To12HourTime(startTime?: HourMinute, endTime?: HourMinute): string | undefined {
+    if (!startTime || !endTime) {
+        return;
+    }
+
+    const postMeridian = endTime.hour >= 12;
+
+    const timeSuffix = postMeridian ? 'PM' : 'AM';
+
+    const formattedStartHour = `${startTime.hour - Number(postMeridian && 12)}`;
+    const formattedEndHour = `${endTime.hour - Number(postMeridian && 12)}`;
```

Comment:
```suggestion
    const pastNoon = endTime.hour > 12;

    const timeSuffix = endTime.hour >= 12 ? 'PM' : 'AM';

    const formattedStartHour = `${startTime.hour - Number(pastNoon && 12)}`;
    const formattedEndHour = `${endTime.hour - Number(pastNoon && 12)}`;

```

The problem with subtracting based on postMeridian (`>= 12`) is that 12:00 PM is correct, and shouldn't be "formatted." 

<img width="862" alt="Screenshot 2023-08-25 at 11 59 16 AM" src="https://github.com/icssc/AntAlmanac/assets/100006999/98f3860c-72a9-454a-864b-4084809b232e">

Note: `pastNoon` is also, in my opinion, not a particularly good name since it doesn't really convey that it's non-inclusive of 12... (?), but `postMeridian` doesn't seem like it would work if it doesn't apply to 12.

Reply:
:sunglasses:

```ts
`${(startTime.hour - Number(pastNoon && 12)}) || 12`;
```

Ok yeah, let's just keep the ternary lol

Reply:
I feel like another conditional is one layer too deep for me 😵‍💫. I think I'll just do the ternary 😅